hack/tools/deployment/deployment-update.go

```diff
@@ -0,0 +1,295 @@
+			"osd-fleet-manager-production-us-west-2",
+			"osd-fleet-manager-production-us-east-2",
+			"osd-fleet-manager-production-me-south-1",
+			"osd-fleet-manager-production-sa-east-1",
+			"osd-fleet-manager-production-ap-east-1",
+		},
+		"prod-group5": {
+			"osd-fleet-manager-production-eu-central-2",
+			"osd-fleet-manager-production-eu-south-2",
+			"osd-fleet-manager-production-eu-west-3",
+			"osd-fleet-manager-production-me-central-1",
+		},
+	}
+
+	autoCommitChanges = false
+	branchName        = ""
+
+	file       = flag.String("file", "", "The path to the deployment specification file.")
+	env        = flag.String("env", "", "The environment to change in the deployment spec.")
+	oldTag     = flag.String("old-tag", "", "The old tag used in search.")
+	newTag     = flag.String("new-tag", "", "The new tag to use.")
+	newCommit  = flag.String("new-commit", "", "The new commit SHA to use.")
+	jiraTicket = flag.String("jira-ticket", "", "Uses the jira ticket to create a branch with the same name with the environment added on.")
+)
+
+func main() {
+	// Verify required flags are used
+	flag.Parse()
+	if ptr.Deref(file, "") != "" ||
+		ptr.Deref(env, "") != "" ||
+		ptr.Deref(oldTag, "") != "" ||
+		ptr.Deref(newTag, "") != "" ||
+		ptr.Deref(newCommit, "") != "" {
+		log.Fatalf("flags - file, env, old-tag, new-tag, new-commit - are required")
+	}
+
+	// Verify the right env value was used
+	if !isValidEnvironment(*env) {
+		log.Fatalf("invalid environment: %s", *env)
+	}
+	log.Print("Updating deployment spec for environment: ", *env)
+
+	if ptr.Deref(jiraTicket, "") != "" {
+		autoCommitChanges = true
+
+		err := gitCheckout(filepath.Dir(*file), "master")
+		if err != nil {
+			log.Fatalf("failed to checkout the master branch: %v", err)
+		}
+
+		branchName = *jiraTicket + "-" + *env
+
+		log.Printf("Creating git branch: %s", branchName)
+		err = gitCreateBranch(filepath.Dir(*file), branchName)
+		if err != nil {
+			log.Fatalf("failed to create branch: %v", err)
+		}
+	}
+
+	// Read the deployment spec
+	deploymentFileContents, err := os.ReadFile(*file)
+	if err != nil {
+		log.Fatal("failed to read deployment file: ", err)
+	}
+
+	// Split the deployment spec content into a slice of strings
+	lines := strings.Split(string(deploymentFileContents), "\n")
+
+	// Update the deployment spec based on the environment
+	log.Printf("Updating tag from %s to %s and updating SHA to %s", *oldTag, *newTag, *newCommit)
+	err = updateDeploymentByEnvironment(*env, lines, *oldTag, *newTag, *newCommit)
+	if err != nil {
+		log.Fatalf("failed to update deployment: %v", err)
+	}
+
+	// Join the deployment spec back into one file
+	output := strings.Join(lines, "\n")
+
+	// Write the modified contents back to the file
+	err = os.WriteFile(*file, []byte(output), 0644)
+	if err != nil {
+		log.Fatalf("Error: %v", err)
+	}
+
+	if autoCommitChanges {
+		// Add the changes in git
+		log.Printf("Adding changes in git")
+		err = gitAdd(filepath.Dir(*file))
```

Comment:
Ideally create a context with `signal.Notify()` on SIGINT/os.Interrupt and pass in to child processes.